tests/chains/Cosmos/CosmosTestHelpers.h

```diff
@@ -0,0 +1,109 @@
+        if (addressParameters.standaloneChain) {
+            const auto hrp_utf8 = STRING(addressParameters.hrp.c_str());
+            const auto data = DATA(addressParameters.publicKey.c_str());
+            const auto pubkey = TWPublicKeyCreateWithData(data.get(), TWPublicKeyTypeSECP256k1);
+            const auto twAddress = TWAnyAddressCreateBech32WithPublicKey(pubkey, TWCoinTypeCosmos, hrp_utf8.get());
+            auto twData = TWAnyAddressData(twAddress);
+            auto hexData = hex(*reinterpret_cast<const Data*>(twData));
+            ASSERT_EQ(hex(Bech32Address(addressParameters.hrp, TW::Hash::HasherSha256ripemd, pubkey->impl).getKeyHash()), hexData);
+            auto address = TWAnyAddressDescription(twAddress);
+            EXPECT_EQ(addressParameters.address, *reinterpret_cast<const std::string*>(address));
+            TWStringDelete(address);
+            TWAnyAddressDelete(twAddress);
+            TWDataDelete(twData);
+            TWPublicKeyDelete(pubkey);
+        }
+
+        // With coin type
+        {
+            auto publicKey = PublicKey(parse_hex(addressParameters.publicKey), addressParameters.publicKeyType);
+            auto address = Address(addressParameters.coinType, publicKey);
+        }
+    }
+
+    static inline void createFromPrivKeyWrapper(const CosmosAddressParameters& addressParameters) {
+        auto privateKey = PrivateKey(parse_hex(addressParameters.privKey));
+        auto address = Address(addressParameters.coinType, privateKey.getPublicKey(addressParameters.publicKeyType));
+        ASSERT_EQ(address.string(), addressParameters.address);
+    }
+
+    static inline void createFromStringWrapper(const CosmosAddressParameters& addressParameters) {
+        // BECH32
+        if (addressParameters.standaloneChain) {
+            const auto address = STRING(addressParameters.address.c_str());
+            const auto hrp = STRING(addressParameters.hrp.c_str());
+            const auto anyAddr = TWAnyAddressCreateBech32(address.get(), TWCoinTypeCosmos, hrp.get());
+            const auto addrDescription = TWAnyAddressDescription(anyAddr);
+            ASSERT_TRUE(TWAnyAddressIsValidBech32(addrDescription, TWCoinTypeCosmos, hrp.get()));
+            TWStringDelete(addrDescription);
+            TWAnyAddressDelete(anyAddr);
+        }
+
+        // With Coin Type
+        {
+            const auto address = STRING(addressParameters.address.c_str());
+            const auto anyAddr = TWAnyAddressCreateWithString(address.get(), addressParameters.coinType);
+            const auto addrDescription = TWAnyAddressDescription(anyAddr);
+            ASSERT_TRUE(TWAnyAddressIsValid(addrDescription, addressParameters.coinType));
+            TWStringDelete(addrDescription);
+            TWAnyAddressDelete(anyAddr);
+        }
+    }
+}
+
+namespace TW::Cosmos::tests {
+    static inline void allAddressTestsWrapper(const CosmosAddressParameters& addressParameters) {
```

Comment:
`Wrapper` is a noun, better to call `TestCosmosAddress` or `TestCosmosAddressParameters` etc

Reply:
Fixed!